fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py

```diff
@@ -61,8 +60,15 @@ def __init__(self, estimator, constraints, eps=0.01, T=50, nu=None, eta_mul=2.0)
         self._T = T
         self._nu = nu
         self._eta_mul = eta_mul
+
         self._best_classifier = None
+        self._best_gap = None
         self._classifiers = None
+        self._weights = None
+        self._last_t = None
+        self._best_t = None
+        self._n_oracle_calls = 0
+        self._oracle_calls_execution_time = None
 
```

Comment:
This __init__ is not [sklearn-compliant,](https://scikit-learn.org/stable/developers/develop.html). Are we leaving that for a future PR?

Reply:
It doesn't strike me as a violation, since there's no logic. But @adrinjalali will have to change a few things with reductions methods anyway, so if this is not compliant we can fix that later (since we aren't compliant yet).